### PR TITLE
Create rules disabled by default

### PR DESCRIPTION
**Pull request by bhirsz**
Relates #809

I have added option to pass ``enabled`` parameter when creating the Rule. It allows us to create rules disabled by default.
The enabled parameter reuses the same concept as ``severity`` and it makes it possible to configure it from CLI too (robocop -c rule_name:enabled:True). 

I have also detected issue that plagued us for quite a long time - the tests that were failing not related to our changes, and only if we run all tests. Turns out when we're loading the rules we're importing them (robocop.checkers.comments.. ) and each test configuration can overwrite the rules attributes stored in sys.modules. I have added a workaround that forcefully removes the already imported rules module if we attempt to load it again (for example when running multiple tests in the same process).

### CODE REVIEW

File: docs/external_rules.rst [hunk 1]
```diff
@@ -148,6 +148,29 @@ Dotted syntax is also supported::
 :code:`rules` dictionary should be available at the same level as checker that is using it. That's why if you are defining your
 external rules using modules and ``__init__.py`` it should be also imported (or defined directly in ``__init__.py``).
 
+Disabled by default rules
+-------------------------
+
+All rules are enabled by default and included after importing them. It is possible to define rule that is disabled
+by default by using ``enabled`` parameter::
```
Comment (mnojek) [inline docs/external_rules.rst:155]:
```suggestion
All rules are enabled by default and included after importing them. It is possible to define a rule that is disabled
by using ``enabled`` parameter with ``False`` value::
```

File: docs/external_rules.rst [hunk 2]
```diff
@@ -148,6 +148,29 @@ Dotted syntax is also supported::
...
+            msg="Custom rule message",
+            severity=RuleSeverity.INFO,
+            enabled=False,
+            docs="""
+            Custom rule description.
+            """,
+        )
+    }
+
+Such rules can be enabled with ``--include`` option or by configuring ``enabled`` parameter directly::
```
Comment (mnojek) [inline docs/external_rules.rst:170]:
```suggestion
Such rules can be enabled when called explicitly with ``--include`` option::

    robocop --include custom-rule .

 or by configuring ``enabled`` parameter directly::
```

File: docs/releasenotes/3.3.0.rst [hunk 3]
```diff
@@ -37,6 +37,32 @@ as in test case files.
 New E0416 ``invalid-setting-in-resource`` rule detects unsupported
 settings in resource files.
 
+Other features
+==============
+
+New ``enabled`` Rule parameter used to create disabled by default rules (#809)
```
Comment (mnojek) [inline docs/releasenotes/3.3.0.rst:43]:
```suggestion
New ``enabled`` rule parameter used to create rules disabled by default (#809)
```

File: docs/external_rules.rst [hunk 4]
```diff
@@ -148,6 +148,29 @@ Dotted syntax is also supported::
 :code:`rules` dictionary should be available at the same level as checker that is using it. That's why if you are defining your
 external rules using modules and ``__init__.py`` it should be also imported (or defined directly in ``__init__.py``).
 
+Disabled by default rules
```
Comment (mnojek) [inline docs/external_rules.rst:151]:
```suggestion
Rules disabled by default
```

File: docs/releasenotes/3.3.0.rst [hunk 5]
```diff
@@ -37,6 +37,32 @@ as in test case files.
...
+------------------------------------------------------------------------------
+
+Previously, all defined rules were enabled by default and could be disabled using ``--exclude`` option (or by
+using ``--include`` if not listed with the option). It is now possible to create rule disabled by default using
```
Comment (mnojek) [inline docs/releasenotes/3.3.0.rst:47]:
```suggestion
using ``--include`` if not listed with the option). It is now possible to create a rule disabled by default using
```

File: docs/releasenotes/3.3.0.rst [hunk 6]
```diff
@@ -37,6 +37,32 @@ as in test case files.
...
+            msg="Custom rule message",
+            severity=RuleSeverity.INFO,
+            enabled=False,
+            docs="""
+            Custom rule description.
+            """,
+        )
+    }
+
+Such rules can be enabled using ``--include`` option or by configuring ``enabled`` parameter directly::
```
Comment (mnojek) [inline docs/releasenotes/3.3.0.rst:63]:
```suggestion
Such rules can be enabled when called explicitly with ``--include`` option::

    robocop --include custom-rule .

 or by configuring ``enabled`` parameter directly::
```

File: robocop/utils/misc.py
Comment (mnojek) [inline robocop/utils/misc.py:1]: I think it makes sense to split these 2 changes into separate PRs - one for ``enabled`` parameter, and one for fixing the issue with Pytest. Is it fine for you to do it?

**Review by mnojek — CHANGES_REQUESTED** [review]

File: robocop/utils/misc.py
Comment (bhirsz) [reply robocop/utils/misc.py:1]: The problem is that this PR will not work without this fix. I could split and merge the fix first, then this PR, is it fine?

Comment (mnojek) [reply robocop/utils/misc.py:1]: It would be better, but if you don't see the value in doing this, I am fine with merging these 2 things in one PR.

**Review by mnojek — APPROVED** [review]
